components/br-stream/src/endpoint.rs

```diff
@@ -23,12 +27,13 @@ pub struct Endpoint<S: MetaStore + 'static> {
     #[allow(dead_code)]
     config: BackupStreamConfig,
     meta_client: Option<MetadataClient<S>>,
-    range_router: Router,
+    range_router: Arc<Mutex<Router>>,
```

Comment:
make this use a read-write lock?

Reply:
Why not make a hashMap(tableID->task) replace BTreeMap ? Thus we can check k-v event being observing More efficiently by extracting tableID from k-v event.

Reply:
well ptal @3pointer about this question ↑

Reply:
I leave a TODO [here](https://github.com/tikv/tikv/blob/9498c241d1d39c26667844fbdfd94c6ccaae5f03/components/br-stream/src/router.rs#L57), I think we can refactor this in future. maybe use multiple kinds of map as trade off. If we only use one hashmap<tableID, taskRange>, we need to handle that a full range backup may include thousands of tables. which may generate too many keys(tableID) points to one task.